### PR TITLE
pkg/trace/api: Reduce flakiness of TestOOMKill

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
Reduce the flakiness of this test - The only good flakes are corn flakes.

### Motivation
This test flaked recently, it seems that under heavy cpu load while running this test it's possible the agent doesn't OOM in time for this test. By increasing the load, lowering the limit and giving a bit of extra deadline we can reduce the flakiness here.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/api/api_oom_test.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ func TestOOMKill(t *testing.T) {
 	conf := config.New()
 	conf.Endpoints[0].APIKey = "apikey_2"
 	conf.WatchdogInterval = time.Millisecond
-	conf.MaxMemory = 0.5 * 1000 * 1000 // 0.5M
+	conf.MaxMemory = 0.4 * 1000 * 1000 // 0.4M
```
Comment (katiehockman) [inline pkg/trace/api/api_oom_test.go:39]: why not just make this really tiny, like a couple KB, to make sure it gets killed as expecte?

Comment (ajgajg1134) [reply pkg/trace/api/api_oom_test.go:39]: I'm not quite sure? My assumption was it was to be at least a "little" sure that the oom kill functionality doesn't activate too quickly?

Comment (katiehockman) [reply pkg/trace/api/api_oom_test.go:39]: Sorry I didn't get back to you here! I forgot about this PR, and then when I noticed the flake, I went ahead and lowered the memory to make sure it triggers: https://github.com/DataDog/datadog-agent/commit/269c9dc9ff5fc0a77f2bdb5056691ec8d5adb464
You make a fair point that we want to make sure that the OOM kill functionality is working well. LMK if you want to close this PR or try to merge it anyway.

**Comment by ajgajg1134** [comment]
closing since this is no longer relevant after https://github.com/DataDog/datadog-agent/pull/18692

**Pull request closed without merging**